src/azure-cli/azure/cli/command_modules/appconfig/_params.py

```diff
@@ -63,7 +63,10 @@ def load_arguments(self, _):
 
     with self.argument_context('appconfig') as c:
         c.argument('resource_group_name', arg_type=resource_group_name_type)
-        c.argument('name', options_list=['--name', '-n'], id_part='None', help='Name of the App Configuration. You can configure the default name using `az configure --defaults app_configuration_store=<name>`', configured_default='app_configuration_store')
+        c.argument('name', options_list=['--name', '-n'], id_part='None',
+                   help='Name of the App Configuration. If the auth mode is "key", you should have "Contributor" role assigned in order to fetch access keys. ' +
+                   'If the auth mode is "login", you should have "Reader" or "Contributor" role assigned in order to fetch store endpoint. ' +
```

Comment:
I don't think we need this explanation. As we have help message for auth_mode and lots of helpful error messages if they misuse this parameter.